### ISSUE TITLE
Bloat - Some Thoughts

### ISSUE THREAD

**Issue by HugoQwerty** [robotex140/Hydrocraft#74]
Should we consider having a big cull of recipes / items to cut down on the number of them?

Some examples, I was going to remove the pumpkin knife, but then I thought why not go further and remove the 3 'Carve Jack-O-Lantern' recipes as well, and the 3 'Lite Jack-O-Lantern'.

There are 12 (!) 'Make Sand' recipes.  Sandstone + Sledgehammer + Sack seems enough to me - and besidles you can just go around the map collecting sand.

Books that give a small mood bonus and can be re-read, just get rid of them?

On the subject of books... given that adding items to a loot table seems to cause more items to spawn - we could do something like: have basic vanilla books spawn only in school desks / lockers, then have higher tier (e.g. 4 & 5) and HC books in bookshops and libraries.  Also have magazines only spawn in magazine racks - including the 'map' and 'newspaper' racks.  Maybe move some books to residential locations but not book shops?

Seed packets, I can replace all the 'open packet' and 'put seeds in packet' recipes (~150+ recipes?) with just two recipes - just use lua to check which packet is being opened and giving the correct reward, and a seed check to determine which seed packet to give.

Jello.  There are Prepare, Pour, & Remove for each type of jello.  I might be in a minority, but I'd just get rid of these.

Useless items, e.g. Dutch Oven and Wok - a Recycle X recipe doesn't seem like a good enough reason to keep them.

Have less 'Recycle' recipes?  E.g. there are Recycle Steel recipe for HCSawcircularblade=3,  HCRotor=2,  HCClutch=2,  HCCamshaft=2,  etc.   I can see a case for Recyce X for things like scrap metal, but bespoke recipes for weird items is just bloat IMO.  There are 78 'Recycle Steel' recipes.

Thoughts?

**Comment by robotex140** [comment]
Yeah one thing that I was thinking, is that we could at some point Rehaul how melting stuff down works. In a way it is already really simplified by using recipes. I was thinking it would be interesting if we could make it work like an oven. Obvs not just trying to use the cooking code, but we would have a container, which you add fuel to and ignite, like how the metal stoves work, and when an item is heated enough, it would turn into molten metal which is just stored in the world object as modData. Then you would use the molten metal to make items. That way we can entirely remove all those items from the recipe system, so that way we remove about 300~probably even more recipes from the crafting menu. I thought that would be a good revamp for the system to be like more actual smelting, and it would significantly improve performance. Although I think we should wait to see what b42 metal working will be like anyways before committing to any big changes

**Comment by Zudakas** [comment]
The molten metal idea is great, I'd love to see that added in. Possibly stored as a "liquid" but only in certain containers (clay jar, item mold, crucible), it would be interesting to pour (currently heated) molten metal into a mold, possibly using the same idea for the open seed packet above to figure out which mold is being used, to create anything from metal sheets and bars to axes and baking trays.

**Comment by ttr** [comment]
recipes - they could be consolidated but they are usually different time to finish based of what tool was used cross-layered with things that you can have bucket of sand and bag of sand. for latter you could have one recipe to "pour sand to bucket and remove all bag ones).
Also, perhaps different approach would be better is to change some devices (like drying rack, Macerator, etc) would not be crafting recipe but operating or passive time (so or like oven, that you put in stuff and let it do it's thing)
And perhaps review those tools -  Macerator, stamp that it's hand, wood and generator based - get it into 1 tool that require electricity.

Look on smithy hammer/anvil/tongs and so on - I do recall there was 'progression' in mind - you start with wood tongs, make sone hammer and stone anvil and go into copper, bronze, iron, and so on - this is not blacksmith rpg - limit it to 2 each and do much shorter progression (so stone one for crude recipes and early gameplay)
My pain here is tongs - cant ever find one and going via this progression make me just not do it.

The scrap ones, i would covert to one per resulting material recipe and add lua code to return scrap adjusted based on item weight (+ skill offset).
Similary to recycle one - combine them to one per material that will result one ingot per item (so input for smelt glass would be 10 bulbs or 1 aquarium or, etc)

On seeds - perhaps remove "put seeds to packet" one as it's quite useless (same as package smokes into box/carton). Also where the package came from, huh? :)

On random/pointless books - personally i never used them but i would keep them. They not in recipes and they create immersive noise - going via bookstore to look up for all needed books they not only pollute loot table but visually annoy you so you need to focus bit more. they are like junk/useless items that come from vanilla game. They are also good tinder in winter :)
